src/runtime/complex.cpp

```diff
@@ -199,25 +228,65 @@ extern "C" Box* complexDiv(BoxedComplex* lhs, Box* rhs) {
 }
 
 Box* complexPos(BoxedComplex* self) {
-    assert(self->cls == complex_cls);
+    if (!isSubclass(self->cls, complex_cls))
+        raiseExcHelper(TypeError, "descriptor '__pos__' requires a 'complex' object but received a '%s'",
+                       getTypeName(self));
     return PyComplex_FromDoubles(self->real, self->imag);
 }
 
 // str and repr
-// For now, just print the same way as ordinary doubles.
-// TODO this is wrong, e.g. if real or imaginary part is an integer, there should
-// be no decimal point, maybe some other differences. Need to dig deeper into
-// how CPython formats floats and complex numbers.
-// (complex_format in Objects/complexobject.c)
-std::string complexFmt(double r, double i, int precision, char code) {
+Box* complexFmt(double r, double i, int precision, char format_code) noexcept {
```

Comment:
Would you mind if we called this `complex_fmt` -- I think it's nice to use CPython's snake_case naming when using their exception style, and use our camelCase naming when using our exceptions.